lib/mozart_fetcher/http_client.ex

```diff
@@ -13,26 +13,36 @@ defmodule HTTPClient do
         hackney: [pool: :origin_pool]
       ]
 
-      case make_request(sanitise(endpoint), headers, options) do
-        {:error, %HTTPoison.Error{reason: "closed"}} ->
-          handle_response(make_request(sanitise(endpoint), headers, options))
-        {k, response} -> handle_response({k, response})
+      case log_errors_and_return(make_request(sanitise(endpoint), headers, options, client)) do
+        {:error, %HTTPoison.Error{reason: :closed}} ->
+          make_request(sanitise(endpoint), headers, options, client)
+
+        {:error, %HTTPoison.Error{reason: :timeout}} ->
+          make_request(sanitise(endpoint), headers, options, client)
+
+        {k, resp} ->
+          {k, resp}
       end
+      |> log_errors_and_return()
```

Comment:
Does this mean we call `log_errors_and_return()` twice? It's also done on line 16

Reply:
Yes. I have just removed one of them in the later PR.